rsv-metrics/libexec/probes/gratia-config-probe

```diff
@@ -268,77 +269,107 @@ EOF
 
   ################################
   # Check last contact time with Gratia server.
-  my $wget_cmdline = shellquote(
-    qw(wget -q --tries=1 --timeout=10 -O -),
-    "http://$soaphost/gratia-administration/monitor-status.html?probename=$probename"
+  my @cur_time= localtime();
+  # localtime gives years from 1900
+  my $year = 1900 + $cur_time[5];
+  # perl uses 0 based months
+  my $month = 1 + $cur_time[4];
+  my $gracchost = "https://gracc.opensciencegrid.org";
+  my $graccindex = sprintf "gracc.osg.raw-%04d.%02d", $year, $month;
+  my $es_query = {  "query" => { 
+                      "query_string" => {
+                        "query" => "ProbeName:\"$probename\" AND EndTime:[now-2d TO now]"
+                      },
+                    },
+                    "sort" => [ 
+                      {
+                        "\@received" =>  {
+                            "order" => "desc",
+                          },
+                      },
+                    ],
+                    "size" => 1
+                 };
+  my $es_json = encode_json $es_query;
+  my $curl_cmdline = shellquote(
+    qw(curl -q --retry 1  --connect-timeout 10 -XGET ),
+    "$gracchost/q/$graccindex/_search",
+    "-d",
+    $es_json
   );
-  my $probe_contact_check = `$wget_cmdline`;
+  my $probe_contact_check = `$curl_cmdline`;
   chomp $probe_contact_check;
-  $probe_contact_check =~ s&^last-contact=([^\s]+)\s+([^\s]+?)\|?$&${1} ${2}&;
-
   # Check to see if we've ever had a contact with this collector
-  if ($probe_contact_check =~ /^last-contact=never/) {
-    setCritical();
-    $run_me_root = RMR_IF_NOT_OK;
-    &appendDetailsData ("ERROR: Probe $probename has never been seen by Gratia collector $soaphost\n");
-    normal_exit();
-  }
-
-  # This means either the collector didn't respond or we've checked in at least once.
-  $probe_contact_check = ParseDate($probe_contact_check);
-  my $last_contact_string = UnixDate($probe_contact_check, "%Y-%m-%d %T %Z");
-  my $cmpDateHour = DateCalc("today", "- 1 hour");
-  my $cmpDateDay = DateCalc("today", "- 1 day");
-  my $probe_seen_by_soaphost;
-  if (!$probe_contact_check) {
+  if ($probe_contact_check =~ /"hits.*?total"\s*:\s*(\d+)/) {
```

Comment:
Thanks @sthapa, the `JSON` addition is great.  Would it be possible to parse `$probe_contact_check` using `decode_json` also?